great_expectations/data_context/types/base.py

```diff
@@ -167,6 +167,7 @@ def __init__(
         splitter_kwargs: Optional[Dict[str, str]] = None,
         sampling_method: Optional[str] = None,
         sampling_kwargs: Optional[Dict[str, str]] = None,
+        reader_options: Optional[Dict[str, Any]] = None,
```

Comment:
As we're now going through schema validation in `list_datasources`, we need to make sure `reader_options` is part of the schema. Unfortunately Marshmallow is quite stringent with `dump` behavior and requires that a particular key be defined in the schema if you want to see it in the output.

Reply:
This is different from `load` which allows you to accept any number of values.